zephyr/lib/alloc.c

```diff
@@ -116,7 +116,7 @@ static inline uintptr_t get_l3_heap_start(void)
 	 * - main_fw_load_offset
 	 * - main fw size in manifest
 	 */
-	return (uintptr_t)z_soc_uncached_ptr((__sparse_force void __sparse_cache *)
+	return (uintptr_t)arch_xtensa_uncached_ptr((__sparse_force void __sparse_cache *)
```

Comment:
Hmm, this seems to go to the wrong direction, zephyr/lib/alloc.c is generic code that should work e.g. on x86 or arm. 

OTOH, many architectures will not support such cached/uncached mappings to same memory, so this will have to:
   - be no-op with architectures with only single addresss (but cache coherency between cores)
   - raise an error on architecture that are have no coherency between cores and do not allow uncached mapping

z_soc_uncached_ptr() seemd like a suitably generic name for such wrapping. Should we add a SOF abstraction for this after all?

Reply:
ack - we would need an ARCH generic Zephyr API for this since it lives in a generic file.

Reply:
see https://github.com/zephyrproject-rtos/zephyr/pull/68140

Reply:
Thanks @nashif - once https://github.com/zephyrproject-rtos/zephyr/pull/68140 is merged, this PR can be updated to use it and we can merge as ARCH generic code.

Reply:
> Thanks @nashif - once [zephyrproject-rtos/zephyr#68140](https://github.com/zephyrproject-rtos/zephyr/pull/68140) is merged, this PR can be updated to use it and we can merge as ARCH generic code.

Ok, Zephyr PR is now merged. @nashif looks like we are good to go with update when you are ready.
@dbaluta this will become arch generic so should build for you on arm64 when the new update lands.